activerecord/lib/active_record/associations/singular_association.rb

```diff
@@ -1,12 +1,24 @@
+require "active_support/deprecation"
+
 module ActiveRecord
   module Associations
     class SingularAssociation < Association #:nodoc:
       # Implements the reader method, e.g. foo.bar for Foo.has_one :bar
-      def reader(force_reload = false)
-        if force_reload && klass
-          klass.uncached { reload }
+      def reader(force_reload = nil, reload: false)
```

Comment:
Shouldn't this `force_reload` parameter be also given the poison object treatment?